### PR TITLE
Refactor to use type hints, Path, OOP and poetry

### PR DESCRIPTION
**Pull request by Dibakarroy1997**
- There was a bug which only allows to scan for .py file till level 2 in the folder structure. This PR uses `Path.rglob` to find all python  files.
- Use type hints
- Extract some methods in different module
- move some methods which uses common attributes in class
- Use poetry

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+keywords = ['snakefood', 'snakefood3', 'dependency graph']
+license = "GPL"
+name = "snakefood3"
+readme = 'README.md'
+repository = 'https://github.com/Trim21/snakefood3'
+version = "0.0.6"
+
+[tool.poetry.dependencies]
+Jinja2 = "^3.1.2"
+python = "^3.8"
```
Comment (Trim21) [inline pyproject.toml:14]: ^3.7

File: snakefood3/utils.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from typing import Generator, Union
+
+
+class Utils:
+    @classmethod
```
Comment (Trim21) [inline snakefood3/utils.py:6]: what's the point adding a extra level namespace?

**Comment by Trim21** [comment]
Thanks for your pr

**Comment by Trim21** [comment]
If you are using black as formatted, add it to dev-depdencies

**Comment by Trim21** [comment]
And I prefer pytest if you want to add testing.

File: snakefood3/utils.py (hunk 2 above)
Comment (Dibakarroy1997) [reply snakefood3/utils.py]: moved it to same class

**Pull request merged**